src/services/monitoring.ts

```diff
@@ -10,422 +21,294 @@ const {
   MonitoringReviewLink,
   MonitoringReviewStatusLink,
   MonitoringClassSummary,
+  MonitoringFindingLink,
+  MonitoringFindingHistory,
+  MonitoringFinding,
+  MonitoringFindingStatusLink,
+  MonitoringFindingStatus,
+  MonitoringFindingStandard,
+  MonitoringStandardLink,
+  MonitoringStandard,
 } = db;
 
-interface IMonitoringReview {
-  reportDeliveryDate: Date;
-  id: number;
-  reviewType: string;
-  statusLink: {
-    status: {
-      name: string;
-    }
-  };
-}
+export async function grantNumbersByRecipientAndRegion(recipientId: number, regionId: number) {
+  const grants = await Grant.findAll({
+    attributes: ['number'],
+    where: {
+      recipientId,
+      regionId,
+    },
+  }) as { number: string }[];
 
-interface IMonitoringReviewGrantee {
-  id: number;
-  grantId: number;
-  reviewId: number;
-  monitoringReviewLink: {
-    monitoringReviews: IMonitoringReview[];
-  }
+  return grants.map((grant) => grant.number);
 }
 
-interface IMonitoringResponse {
-  recipientId: number;
-  regionId: number;
-  reviewStatus: string;
-  reviewDate: string;
-  reviewType: string;
-  grant: string;
-}
+export async function ttaByReviews(
+  recipientId: number,
+  regionId: number,
+): Promise<ITTAByReviewResponse[]> {
+  const grantNumbers = await grantNumbersByRecipientAndRegion(recipientId, regionId) as string[];
+  const reviews = await MonitoringReview.findAll({
+    include: [
```

Comment:
Very close, also need to filter by MonitoringReviewStatuses complete, and only reviews 1/1/2023 or newer for the report delivery date

Reply:
What would be your thoughts on an ssdi solution to this?

Reply:
Not sure what that gets us, this is a little easier to read/maintain to me

Reply:
i am fine with not using ssdi, but the date needs to be in the join/filter logic